testsuite/MDAnalysisTests/formats/test_libmdaxdr.py

```diff
@@ -21,376 +21,485 @@
+                assert_array_almost_equal(box, box_compare, decimal=3)
+                assert step == i
+                assert time == i * 0.5
+                assert prec == 1000.0
+
+
+def test_write_prec(tmpdir):
     mf_xtc = XTCFile(XTC_multi_frame)
-    with XTCFile('foo.xtc', 'w') as f_out:
-        assert_equal(f_out.n_atoms, 0)
-        frame = mf_xtc.read()
-        f_out.write(frame.x, frame.box, frame.step,
-                    frame.time, 100.0)
-
-    xtc = XTCFile('foo.xtc')
-    assert_equal(len(xtc), 1)
-    frame = xtc.read()
-    assert_equal(frame.prec, 100.0)
-
+    with tmpdir.as_cwd():
+        with XTCFile('foo.xtc', 'w') as f_out:
+            assert f_out.n_atoms == 0
+            frame = mf_xtc.read()
+            f_out.write(frame.x, frame.box, frame.step, frame.time, 100.0)
 
-class Test_TRRFile(XDRFormatBaseTest):
-    xdrfile = TRRFile
-    single_frame = TRR_single_frame
-    multi_frame = TRR_multi_frame
-    offsets = np.array([0, 480, 960, 1440, 1920, 2400, 2880,
-                        3360, 3840, 4320])
-
-
-def test_trr():
-    f = TRRFile(TRR_single_frame)
-    assert_equal(f.n_atoms, 10)
-    frame = f.read()
-    # trr only saves with 3 decimal places precision
-    assert_array_almost_equal(frame.x.flat, np.arange(30))
-    assert_array_almost_equal(frame.v.flat, np.arange(30))
-    assert_array_almost_equal(frame.f.flat, np.arange(30))
-    assert_array_almost_equal(frame.box, np.eye(3) * 20)
-    assert_equal(frame.step, 10)
-    assert_equal(frame.time, 5.0)
-    assert_equal(frame.lmbda, 0.5)
+        xtc = XTCFile('foo.xtc')
+        assert len(xtc) == 1
+        frame = xtc.read()
+        assert frame.prec == 100.0
 
 
 def test_read_multi_frame_trr():
     with TRRFile(TRR_multi_frame) as f:
         f.seek(5)
-        assert_equal(f.tell(), 5)
-        assert_equal(len(f), 10)
-        assert_equal(f.tell(), 5)
+        assert f.tell() == 5
+        assert len(f) == 10
+        assert f.tell() == 5
         f.seek(0)
-        assert_equal(f.tell(), 0)
+        assert f.tell() == 0
         ones = np.ones(30).reshape(10, 3)
         box_compare = np.eye(3) * 20
         for i, frame in enumerate(f):
             assert_array_almost_equal(frame.x, ones * i)
             assert_array_almost_equal(frame.v, ones * i + 10)
             assert_array_almost_equal(frame.f, ones * i + 20)
             assert_array_almost_equal(frame.box, box_compare)
-            assert_equal(frame.step, i)
-            assert_equal(frame.time, i * 0.5)
+            assert frame.step == i
+            assert frame.time == i * 0.5
             assert_almost_equal(frame.lmbda, .01 * i)
 
 
-@run_in_tempdir()
-def test_write_trr():
-    with TRRFile(TRR_multi_frame) as f_in, TRRFile('foo.trr', 'w') as f_out:
-        assert_equal(f_out.n_atoms, 0)
-        for frame in f_in:
+@pytest.mark.parametrize('dtype', (np.int32, np.int64, np.float32, np.float64, int, float))
+def test_write_trr_dtype(tmpdir, dtype):
```

Comment:
then we have `def test_a(in_tmpdir):` which runs even more magically in a temporary directory and we don't use the `in_tmpdir` variable. Not using a variable isn't so nice since I would like to enable pylint to warn about unused variables soon.

So good idea. But maybe to much magic